tests/test_sealapi.py

```diff
@@ -0,0 +1,442 @@
+    assert sealapi.Serialization.IsSupportedComprMode(15) == False
+    assert sealapi.Serialization.IsSupportedComprMode(0) == True
+
+    assert sealapi.Serialization.ComprSizeEstimate(
+        256, sealapi.COMPR_MODE_TYPE.NONE) == 256
+
+    header = sealapi.Serialization.SEALHeader()
+    assert sealapi.Serialization.IsValidHeader(header) == True
+
+    header.magic = 0
+    assert sealapi.Serialization.IsValidHeader(header) == False
+
+
+def test_plaintext():
+    testcase = sealapi.Plaintext()
+    assert testcase.coeff_count() == 0
+
+    testcase = sealapi.Plaintext(15)
+    assert testcase.coeff_count() == 15
+
+    testcase = sealapi.Plaintext(100, 15)
+    assert testcase.coeff_count() == 15
+    assert testcase.capacity() == 100
+
+    testcase = sealapi.Plaintext("7FFx^3 + 1x^1 + 3")
+    assert testcase.coeff_count() == 4
+    assert testcase.significant_coeff_count() == 4
+    assert testcase.capacity() == 4
+
+    testcase2 = testcase
+    assert testcase.coeff_count() == 4
+    assert testcase.capacity() == 4
+
+    testcase = sealapi.Plaintext(100, 15)
+    testcase.reserve(200)
+    assert testcase.capacity() == 200
+
+    testcase = sealapi.Plaintext("7FFx^3 + 1x^1 + 3")
+    assert testcase.capacity() == 4
+    testcase.reserve(200)
+    assert testcase.capacity() == 200
+    testcase.shrink_to_fit()
+    assert testcase.capacity() == 4
+
+    testcase.release()
+    assert testcase.coeff_count() == 0
+
+    testcase = sealapi.Plaintext("7FFx^3 + 1x^1 + 3")
+    assert testcase.coeff_count() == 4
+    assert testcase.nonzero_coeff_count() == 3
+    testcase.resize(10)
+    assert testcase.coeff_count() == 10
+
+    testcase.set_zero()
+    assert testcase.is_zero()
+    assert testcase.nonzero_coeff_count() == 0
+
+    testcase = sealapi.Plaintext("7FFx^3 + 2x^1 + 3")
+    assert testcase.to_string() == "7FFx^3 + 2x^1 + 3"
+
+
+def test_randomgen():
+    assert sealapi.random_uint64() != sealapi.random_uint64()
+
+    seed = [sealapi.random_uint64() for i in range(8)]
+
+    for factory in [
+            sealapi.BlakePRNGFactory.DefaultFactory(),
+            sealapi.BlakePRNGFactory(),
+            sealapi.BlakePRNGFactory(seed)
+    ]:
+        for generator in [
+                factory.create(),
+                factory.create([sealapi.random_uint64() for i in range(8)])
+        ]:
+
+            assert generator.generate() != generator.generate()
+
+            for i in range(1024):
+                generator.refresh()
+                generator.generate()
+
+
+def test_encryptionparams():
+    schemes = [
+        sealapi.SCHEME_TYPE.NONE, sealapi.SCHEME_TYPE.BFV,
+        sealapi.SCHEME_TYPE.CKKS
+    ]
+
+    for scheme_id, scheme in enumerate(schemes):
+        assert int(scheme) == scheme_id
+
+        testcase = sealapi.EncryptionParameters(scheme)
+        assert testcase.scheme() == scheme
+
+        testcase = sealapi.EncryptionParameters(scheme_id)
+        assert testcase.scheme() == scheme
```

Comment:
You can use pytest.mark.parametrize when you need to do similar test on different values.